### PR TITLE
Refactor a few CSS properties

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17095)

### CODE REVIEW

**Comment by nox** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=d3ab8b33176a3b58691303a84d9b1d8a30ede14e

File: components/style/values/specified/transform.rs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            input.try(|i| Length::parse(context, i)).unwrap_or(Length::from_px(0.))
+        };
+        match input.try(|i| OriginComponent::parse(context, i)) {
+            Ok(x_origin @ OriginComponent::Center) => {
+                if let Ok(y_origin) = input.try(|i| OriginComponent::parse(context, i)) {
+                    let depth = parse_depth(input);
+                    return Ok(Self::new(x_origin, y_origin, depth));
+                }
+                let y_origin = OriginComponent::Center;
+                if let Ok(x_origin) = input.try(|i| OriginComponent::parse(context, i)) {
```
Comment (emilio) [inline components/style/values/specified/transform.rs]: How can this succeed now if it failed in the branch above? Am I missing something obvious here?

**Comment by jdm** [comment]
@bors-servo: retry
* ssl timeout

**Pull request merged**